pango/src/script_iter.rs

```diff
@@ -0,0 +1,173 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::{marker::PhantomData, mem, ptr};
+
+use glib::{translate::*, GStr};
+
+use crate::Script;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ScriptIter<'text> {
+    ptr: ptr::NonNull<ffi::PangoScriptIter>,
+    list: PhantomData<&'text crate::AttrList>,
+}
+
+#[cfg(any(feature = "v1_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v1_44")))]
+impl<'text> Clone for ScriptIter<'text> {
+    #[inline]
+    fn clone(&self) -> Self {
+        let ptr = unsafe {
+            ptr::NonNull::new_unchecked(glib::gobject_ffi::g_boxed_copy(
+                ffi::pango_script_iter_get_type(),
+                self.ptr.as_ptr() as *mut _,
+            ) as *mut _)
+        };
+        Self {
+            ptr,
+            list: PhantomData,
+        }
+    }
+}
+
+impl<'text> Drop for ScriptIter<'text> {
+    #[inline]
+    fn drop(&mut self) {
+        unsafe {
+            ffi::pango_script_iter_free(self.ptr.as_ptr());
+        }
+    }
+}
+
+#[cfg(any(feature = "v1_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v1_44")))]
+impl<'text> glib::StaticType for ScriptIter<'text> {
+    #[inline]
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::pango_script_iter_get_type()) }
+    }
+}
+
+impl<'text> ScriptIter<'text> {
+    #[doc(alias = "pango_script_iter_new")]
+    pub fn new(text: impl AsRef<GStr> + 'text) -> Self {
```

Comment:
Do we want to change this here to `AsRef<GStr>` already, or defer that to next release with the other related changes?

Note: only for parameters this is potentially problematic

Reply:
I think this one should stay `AsRef<GStr>` for now, with an owned `GString` it forces the `Iterator` implementation to copy the string every time.

With `IntoGStr` we can fix this and also optimize out those `GStr::from_ptr` calls.

Reply:
Works for me